src/Mod/Sketcher/Gui/DrawSketchHandler.cpp

```diff
@@ -454,8 +455,9 @@ int DrawSketchHandler::seekAutoConstraint(std::vector<AutoConstraint>& suggested
         if (geom) {
             GeoId = preSelCrv;
             if (geom->is<Part::GeomLineSegment>()) {
-                const Part::GeomLineSegment* line = static_cast<const Part::GeomLineSegment*>(geom);
+                auto* line = static_cast<const Part::GeomLineSegment*>(geom);
                 hitShapeDir = line->getEndPoint() - line->getStartPoint();
+                preselectIsLine = true;

```

Comment:
Can you briefly explain why adding a check for line helps us decide for tangents of circles?

Reply:
The if block that you modified, was intended only for lines (as the comment mention). `hitShapeDir` is defined only for lines. So for circles it is 0 length.

The if block was
`if (constr.Type == Sketcher::Tangent && Dir.Length() > 1e-8 && hitShapeDir.Length() > 1e-8) {`
So circles was not triggering this block. So it circle case was going to `suggestedConstraints.push_back(constr);` adding the cstr.

After your change : 
```
        if (constr.Type == Sketcher::Tangent) {
            if (Dir.Length() < 1e-8 || hitShapeDir.Length() < 1e-8) {
                return suggestedConstraints.size();
            }
```
The circle case was entering in this if block. Then because `hitShapeDir `length was 0 it was returning.